datafusion/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -571,9 +577,9 @@ impl SortPreservingMergeStream {
             let _timer = elapsed_compute.timer();
 
             match self.min_heap.pop() {
-                Some(cursor) => {
+                Some(mut cursor) => {
                     let stream_idx = cursor.stream_idx;
-                    let cursor_idx = self.cursors[stream_idx].len() - 1;
+                    let cursor_idx = self.batches[stream_idx].len() - 1;
```

Comment:
I think this and the corresponding field in `RowIndex` should be updated, perhaps to batch_idx